test/e2e/unmanaged_dns_test.go

```diff
@@ -240,8 +263,35 @@ func TestUnmanagedDNSToManagedDNSInternalIngressController(t *testing.T) {
 		t.Fatalf("failed to update ingresscontroller %s: %v", name, err)
 	}
 
-	if err := kclient.Delete(context.TODO(), lbService); err != nil && !errors.IsNotFound(err) {
-		t.Fatalf("failed to delete svc %s: %v", lbService.Name, err)
+	var oldLoadBalancerStatus corev1.LoadBalancerStatus
+	lbService.Status.LoadBalancer.DeepCopyInto(&oldLoadBalancerStatus)
+
+	// Only delete the service on platforms that don't automatically update the service's scope.
+	switch platform {
+	case configv1.AlibabaCloudPlatformType, configv1.AWSPlatformType, configv1.IBMCloudPlatformType, configv1.PowerVSPlatformType:
+		if err := kclient.Delete(context.TODO(), lbService); err != nil && !errors.IsNotFound(err) {
```

Comment:
Should we have multiple attempts here to mitigate transient failures? Somebody else updated the object, for example.

Reply:
Hmm, that's a good, but loaded question. It's never been an issue in the multiple times I have tried it. Part of me agrees with you, but the other part says "Well, if we do it here, we need to rewrite a ton of other Gets/Deletes/Updates in this e2e-test and others" as this doesn't just apply here.

I think we need standardized common wrapper functions for Delete/Get/Update if we want to mitigate this, opposed as doing it as 1-offs everywhere. Could have a standard "deleteAndRetry" function (I think we have updateAndRetry-type functions already?).

I think probably a good shift-week-type refactoring, I'm hesitant to start doing 1-off-type refactoring since it's in many more than this one place here, and I think the effort should be coordinated. Let me know if you disagree, or have an existing function I can use.

Reply:
+1 to what Grant said.  We've added retries to E2E tests _ad hoc_ in the past, and absent any _hoc_, it's reasonable to defer that to some more comprehensive effort to add retries to Kube API calls in E2E tests.